components/datetime/src/scaffold/calendar.rs

```diff
@@ -355,6 +355,22 @@ pub(crate) struct AnyCalendarForFormatting {
 }
 
 impl AnyCalendarForFormatting {
+    pub(crate) fn from_cldr_calendar(calendar: impl CldrCalendar + IntoAnyCalendar) -> Self {
+        let any_calendar = calendar.to_any();
+        Self {
+            kind: AnyCalendarForFormattingKind::try_from_any_calendar_kind(any_calendar.kind())
+                .unwrap_or_else(|| {
+                    debug_assert!(
+                        false,
+                        "CldrCalendar returns unsupported AnyCalendarKind: {:?}",
+                        any_calendar.kind()
+                    );
+                    AnyCalendarForFormattingKind::Coptic
+                }),
+            any_calendar,
```

Comment:
This is not correct because `JapaneseExtended` can implement `CldrCalendar` and yet not be included in `AnyCalendarForFormattingKind`

Reply:
Right. When we break the link between `AnyCalendarForFormattingKind` and `CldrCalendar` we need to introduce a new trait for `into_formatter` to accept. `into_formatter` can definitely be infallible though, as we statically know the calendar.

Reply:
_I guess_ we can have another scaffolding trait. `CldrCalendarWithDateTimeFormatterSupport`

I really would like to not link CldrCalendar and AnyCalendarForFormattingKind in this PR though, because they are fully decoupled right now. You can add the new trait and push it if you want